Assets/Scripts/Internal/DaggerfallBillboardSystem.cs

```diff
@@ -0,0 +1,173 @@
+                }
+                pmScheduleJobs.End();
+            }
+        }
+
+        public static void AddPointBillboard(Transform t)
+        {
+            int index = pointBillboards.length;
+            pointBillboards.Add(t);
+            pointIndices.Add(t, index);
+        }
+
+        public static void RemoveBillboard(Transform t)
+        {
+#if UNITY_EDITOR
+            // TransformAccessArray should always exists, *except* when in-editor game just stopped playing, hence this null-check.
+            if (!axialBillboards.isCreated || !pointBillboards.isCreated) return;
+#endif
+
+            if (pointIndices.ContainsKey(t))
+            {
+                int index = pointIndices[t];
+                pointBillboards.RemoveAtSwapBack(index);
+                pointIndices.Remove(t);
+
+                int indexThatGotReplaced = pointBillboards.length;
+                var found = pointIndices.FirstOrDefault((next) => next.Value == indexThatGotReplaced);
+                if (found.Key) pointIndices[found.Key] = index;
+            }
+            else if (axialIndices.ContainsKey(t))
+            {
+                int index = axialIndices[t];
+                axialBillboards.RemoveAtSwapBack(index);
+                axialIndices.Remove(t);
+
+                int indexThatGotReplaced = axialBillboards.length;
+                var found = axialIndices.FirstOrDefault((next) => next.Value == indexThatGotReplaced);
+                if (found.Key) axialIndices[found.Key] = index;
+            }
+        }
+
+        public static void AddAxialBillboard(Transform t)
+        {
+            int index = axialBillboards.length;
+            axialBillboards.Add(t);
+            axialIndices.Add(t, index);
+        }
+
+        [Unity.Burst.BurstCompile]
+        public struct SetRotationJob : IJobParallelForTransform
+        {
+            public quaternion Value;
+            void IJobParallelForTransform.Execute(int index, TransformAccess transform)
+                => transform.rotation = Value;
+        }
+
+        [Unity.Burst.BurstCompile]
+        public struct LookAtPointJob : IJobParallelForTransform
+        {
+            public float3 Point;
+            void IJobParallelForTransform.Execute(int index, TransformAccess transform)
+                => transform.rotation = quaternion.LookRotationSafe(Point - (float3)transform.position, new float3 { y = 1 });
+        }
+
+        [Unity.Burst.BurstCompile]
+        public struct AxialLookAtPointJob : IJobParallelForTransform
+        {
+            public float3 Point;
+            void IJobParallelForTransform.Execute(int index, TransformAccess transform)
+            {
+                float3 position = (float3)transform.position;
+                transform.rotation = quaternion.LookRotationSafe(
+                    new float3 { x = Point.x, y = position.y, z = Point.z } - position,
+                    new float3 { y = 1 }
+                );
+            }
+        }
+
```

Comment:
**Sidenote**: `AxialLookAtPointJob` introduces an alternative axial alignment method. Just something to consider or just delete if deemed not worthy.

To test it locally just comment/uncomment these 2 lines: https://github.com/Interkarma/daggerfall-unity/pull/2389/files#diff-61902a9d59619dfbe36dfe1b492f7eeb4192bce7eb5f336d20d61f682fa4f806R81-R82

> `vanilla` left,  `AxialLookAtPointJob` right
<p float="center">
    <img src="https://user-images.githubusercontent.com/3066539/179261004-e7693fa9-f620-456d-9431-6e480ef161cc.png" width="49%" height="300px">
    <img src="https://user-images.githubusercontent.com/3066539/179261013-98e40cdc-c2a0-4a62-9a05-f39ff7f3d401.png" width="49%" height="300px">
</p>

> `vanilla` left,  `AxialLookAtPointJob` right
<p float="center">
    <img src="https://user-images.githubusercontent.com/3066539/179261578-4b6c9c37-8dd7-4586-974c-f548fee71bbe.png" width="49%" height="300px">
    <img src="https://user-images.githubusercontent.com/3066539/179261572-afa97f16-e08e-4677-b003-71fbc130ace4.png" width="49%" height="300px">
</p>

I am not sure which one is better, seems rather arbitrary, just something to consider maybe.

Reply:
Daggerfall's billboards should align to camera facing as they do in classic and in DFU. They both have issues depending on camera alignment and where player is standing, but I find camera-facing to be less obnoxious than billboards spinning around just by walking closer to them. It also matches classic, and I don't wish to change the current behaviour.

Reply:
I understand, [8a50def](https://github.com/Interkarma/daggerfall-unity/pull/2389/commits/8a50deff0992dfa5cc24e24a0061a34fc6502686) reverted this already